extensions/vscode/webviews/homeView/src/HostConduitService.ts

```diff
@@ -196,8 +197,14 @@ const onSaveSelectionMsg = () => {
 };
 
 const onRefreshFilesMsg = (msg: RefreshFilesMsg) => {
-  const home = useHomeStore();
-  home.files = msg.content.files;
+  const fileStore = useFileStore();
+
+  // If the root file has changed, reset the expanded directories
+  if (msg.content.files.abs !== fileStore.files?.abs) {
+    fileStore.expandedDirs = new Set();
+  }
```

Comment:
The expanded directories only reset when the root changes; meaning switching deployments that have the same directory won't reset it.

This feels like good behavior to me, but I want to highlight it for discussion.